cpp/src/arrow/filesystem/s3fs.h

```diff
@@ -152,6 +152,10 @@ struct ARROW_EXPORT S3Options {
 
   S3Options();
 
+  /// Get a stock AWS retry strategy from a string.
+  static std::shared_ptr<S3RetryStrategy> GetS3RetryStrategy(const std::string& name,
```

Comment:
This should probably be a static method of `S3RetryStrategy` instead. Also, the `name` parameter should be explained in the docstring.

Reply:
Also, instead of passing a made-up name, I would favour two separate methods `GetAwsDefaultRetryStrategy` and `GetAwsStandardRetryStrategy`.